### PR TITLE
chore(ui): migrate DataGrid components to TypeScript

### PR DESCRIPTION
**Pull request by gjaskiewicz**
# Summary

Migrate DataGrid components to TypeScript

# Changes Made

- Migrate DataGrid components to TypeScript
- Cut out few deps on Button, Stack, Spinner, Search from stories (to be re-introduced)

# Related Issues

- Issue 1: [https://github.com/cloudoperators/juno/issues/224]

# Testing Instructions

1. `npm i`
2. `turbo test`

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have made corresponding changes to the documentation (if applicable).
- [ ] My changes generate no new warnings or errors.

### CODE REVIEW

**Review by barsukov — CHANGES_REQUESTED** [review]
Please have a look to few types that could be potentially fixed

File: packages/ui-components/src/components/DataGridFoot/DataGridFoot.component.tsx [hunk 1]
```diff
@@ -4,19 +4,18 @@
...
       {children}
     </tfoot>
   )
 }
 
-DataGridFoot.propTypes = {
+export interface DataGridFootProps {
   /** Children to render in the DataGridFoot */
-  children: PropTypes.node,
+  children?: React.ReactNode
```
[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/DataGridFoot/DataGridFoot.component.tsx]: Done

File: packages/ui-components/src/components/DataGridHeadCell/DataGridHeadCell.component.tsx [hunk 2]
```diff
@@ -50,15 +49,15 @@ export const DataGridHeadCell = forwardRef(
...
   // sortable: PropTypes.bool,
   /** Add a col span to the cell. This works like a colspan in a normal html table, so you have to take care not to place too many cells in a row if some of them have a colspan.  */
-  colSpan: PropTypes.number,
+  colSpan?: number
   /** Set nowrap to true if the cell content shouldn't wrap (this is achieved by adding white-space: nowrap;) */
-  nowrap: PropTypes.bool,
+  nowrap?: boolean
   /** Children to render in the DataGridHeadCell */
-  children: PropTypes.node,
+  children?: React.ReactNode
```
[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/DataGridHeadCell/DataGridHeadCell.component.tsx]: Done

File: packages/ui-components/src/components/DataGridToolbar/DataGridToolbar.component.tsx [hunk 3]
```diff
@@ -29,11 +28,11 @@ export const DataGridToolbar = ({ search, className = "", children = null, ...pr
   )
 }
 
-DataGridToolbar.propTypes = {
+export interface DataGridToolbarProps {
   /** Pass an optional SearchInput component */
-  search: PropTypes.node,
+  search?: React.ReactNode
```
[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/DataGridToolbar/DataGridToolbar.component.tsx]: Done

File: packages/ui-components/src/components/DataGridToolbar/DataGridToolbar.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+export default {
+  title: "Components/DataGrid/DataGridToolbar",
+  component: DataGridToolbar,
+  argTypes: {
+    children: {
+      control: false,
+    },
+    search: {
+      control: false,
```
[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/DataGridToolbar/DataGridToolbar.stories.tsx]: Done

File: packages/ui-components/src/components/DataGridFoot/DataGridFoot.component.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (gjaskiewicz) [reply packages/ui-components/src/components/DataGridFoot/DataGridFoot.component.tsx]: Done

**Review by barsukov — APPROVED** [review]

**Review by andypf — APPROVED** [review]

**Pull request merged**